### PR TITLE
MGMT-16507: Fix incorrect reference to ImageStore

### PR DESCRIPTION
**Pull request by paul-maidment**
During the implementation of [MGMT-16507](https://issues.redhat.com//browse/MGMT-16507) the incorrect CA cert was passed into the ImageStore This PR corrects that by fixing the reference

## Assignees

/cc @
/cc @

## Checklist

- [x] Title and description added to both, commit and PR
- [x] Relevant issues have been associated
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [] Does this change include unit tests (note that code changes require unit tests)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func main() {
 		}
 	}
 
-	is, err := imagestore.NewImageStore(isoeditor.NewEditor(Options.DataDir), Options.DataDir, Options.ImageServiceBaseURL, Options.InsecureSkipVerify, versions, Options.AssistedServiceApiTrustedCAFile)
+	is, err := imagestore.NewImageStore(isoeditor.NewEditor(Options.DataDir), Options.DataDir, Options.ImageServiceBaseURL, Options.InsecureSkipVerify, versions, Options.OSImageDownloadTrustedCAFile)
```
Comment (carbonin) [inline main.go:85]: Should we also pass `AssistedServiceApiTrustedCAFile` when creating the assisted service client?

Comment (paul-maidment) [reply main.go:85]: yes, agreed, I have changed this.

**Review by carbonin — APPROVED** [review]

**Comment by carbonin** [comment]
@paul-maidment can you actually look into this failure. It seems related:

```
{"file":"/go/src/github.com/openshift/origin/internal/handlers/iso.go:50","func":"github.com/openshift/assisted-image-service/internal/handlers.(*isoHandler).ServeHTTP","level":"error","msg":"Error retrieving ignition content: Get \"https://assisted-service.assisted-installer.svc:8090/api/assisted-install/v2/infra-envs/9be7b23a-e0bb-470d-9397-9179cf56b2d7/downloads/files?api_key=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiI5YmU3YjIzYS1lMGJiLTQ3MGQtOTM5Ny05MTc5Y2Y1NmIyZDcifQ.4PH_Y5VZ3sAK0wEYIvlnWcRwan6IQLEBOEtJ1GjjBPX5idnH6STGOk9z8AV61d3JvQOoQVh6L__q9BnJ8wKWgw\u0026discovery_iso_type=minimal-iso\u0026file_name=discovery.ign\": tls: failed to verify certificate: x509: certificate signed by unknown authority\n","time":"2024-01-30T15:19:08Z"}
{"file":"/go/src/github.com/openshift/origin/internal/handlers/iso.go:50","func":"github.com/openshift/assisted-image-service/internal/handlers.(*isoHandler).ServeHTTP","level":"error","msg":"Error retrieving ignition content: Get \"https://assisted-service.assisted-installer.svc:8090/api/assisted-install/v2/infra-envs/9be7b23a-e0bb-470d-9397-9179cf56b2d7/downloads/files?api_key=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiI5YmU3YjIzYS1lMGJiLTQ3MGQtOTM5Ny05MTc5Y2Y1NmIyZDcifQ.4PH_Y5VZ3sAK0wEYIvlnWcRwan6IQLEBOEtJ1GjjBPX5idnH6STGOk9z8AV61d3JvQOoQVh6L__q9BnJ8wKWgw\u0026discovery_iso_type=minimal-iso\u0026file_name=discovery.ign\": tls: failed to verify certificate: x509: certificate signed by unknown authority\n","time":"2024-01-30T15:20:51Z"}
{"file":"/go/src/github.com/openshift/origin/internal/handlers/iso.go:50","func":"github.com/openshift/assisted-image-service/internal/handlers.(*isoHandler).ServeHTTP","level":"error","msg":"Error retrieving ignition content: Get \"https://assisted-service.assisted-installer.svc:8090/api/assisted-install/v2/infra-envs/9be7b23a-e0bb-470d-9397-9179cf56b2d7/downloads/files?api_key=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiI5YmU3YjIzYS1lMGJiLTQ3MGQtOTM5Ny05MTc5Y2Y1NmIyZDcifQ.4PH_Y5VZ3sAK0wEYIvlnWcRwan6IQLEBOEtJ1GjjBPX5idnH6STGOk9z8AV61d3JvQOoQVh6L__q9BnJ8wKWgw\u0026discovery_iso_type=minimal-iso\u0026file_name=discovery.ign\": tls: failed to verify certificate: x509: certificate signed by unknown authority\n","time":"2024-01-30T15:24:53Z"}
{"file":"/go/src/github.com/openshift/origin/internal/handlers/iso.go:50","func":"github.com/openshift/assisted-image-service/internal/handlers.(*isoHandler).ServeHTTP","level":"error","msg":"Error retrieving ignition content: Get \"https://assisted-service.assisted-installer.svc:8090/api/assisted-install/v2/infra-envs/9be7b23a-e0bb-470d-9397-9179cf56b2d7/downloads/files?api_key=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiI5YmU3YjIzYS1lMGJiLTQ3MGQtOTM5Ny05MTc5Y2Y1NmIyZDcifQ.4PH_Y5VZ3sAK0wEYIvlnWcRwan6IQLEBOEtJ1GjjBPX5idnH6STGOk9z8AV61d3JvQOoQVh6L__q9BnJ8wKWgw\u0026discovery_iso_type=minimal-iso\u0026file_name=discovery.ign\": tls: failed to verify certificate: x509: certificate signed by unknown authority\n","time":"2024-01-30T15:30:45Z"}
```

File: main.go [hunk 2]
```diff
@@ -48,19 +48,19 @@ var Options struct {
...
-	if Options.HTTPSCAFile != "" {
-		Options.AssistedServiceApiTrustedCAFile = Options.HTTPSCAFile
-	}
 	log.SetReportCaller(true)
 	log.SetFormatter(&log.JSONFormatter{})
 	err := envconfig.Process("cluster-image", &Options)
 	if err != nil {
 		log.Fatalf("Failed to process config: %v\n", err)
 	}
+	if Options.HTTPSCAFile != "" {
```
[the remark this answers is not in the document]
Comment (paul-maidment) [reply main.go]: Good point, fixed now!

**Review by carbonin — APPROVED** [review]

**Review by carbonin — APPROVED** [review]

**Pull request merged**